posteriors/utils.py

```diff
@@ -230,6 +244,180 @@ def fisher(*args, **kwargs):
     return fisher
 
 
+def ggnvp(
+    forward: Callable,
+    loss: Callable,
+    primals: tuple,
+    tangents: tuple,
+    forward_has_aux: bool = False,
+    loss_has_aux: bool = False,
+    normalize: bool = False,
+) -> (
+    Tuple[float, TensorTree]
+    | Tuple[float, TensorTree, Any]
+    | Tuple[float, TensorTree, Any, Any]
+):
+    """Generalised Gauss-Newton vector product.
+
+    Equivalent to the (non-empirical) Fisher vector product when `loss` is the negative
+    log likelihood of an exponential family distribution as a function of its natural
+    parameter.
+
+    Defined as
+    $$
+    G(θ) = J_f(θ) H_l(z) J_f(θ)^T
+    $$
+    where $z = f(θ)$ is the output of the forward function $f$ and $l(z)$
+    is a loss function with scalar output.
```

Comment:
this is not the same as for `fvp`, we should add an example of a gotcha somewhere to flag this

Reply:
I think an example for loss and forward in the doc would be useful, similar to the optax code https://github.com/google-deepmind/optax/blob/test_608993041/optax/second_order/_oracles.py